xmodule/video_block/video_block.py

```diff
@@ -472,26 +478,69 @@ def get_html(self, view=STUDENT_VIEW, context=None):  # lint-amnesty, pylint: di
             'track': track_url,
             'transcript_download_format': transcript_download_format,
             'transcript_download_formats_list': self.fields['transcript_download_format'].values,  # lint-amnesty, pylint: disable=unsubscriptable-object
-            # provide the video url iif the video is public and we are in LMS.
-            # Reverse render_public_video_xblock is not available in studio.
-            'public_video_url': self._get_public_video_url(),
         }
 
+        # Public video previewing / social media sharing
+        if self.is_public_sharing_enabled():
+            template_context['public_sharing_enabled'] = True
+            template_context['public_video_url'] = self.get_public_video_url()
+
         return self.runtime.service(self, 'mako').render_template('video.html', template_context)
 
-    def _get_public_video_url(self):
+    def get_course_video_sharing_override(self):
         """
-        Returns the public video url
+        Return course video sharing options override or None
+        """
+        course = get_course_by_id(self.course_id)
+        course_video_sharing_option = course.video_sharing_options \
+            if hasattr(course, 'video_sharing_options') else None
```

Comment:
if None then do as `per-video`? Is the safety check needed when course.video_sharing_options has a default value already?

Reply:
I do want to preserve whether the course has a value set versus the "per-video" setting. It isn't important now but may be at some point in the future. Unless, is `video_sharing_options` always guaranteed to be set? What about courses that haven't touched this value yet? Do we backfill?

Reply:
I was under the assumption that `video_sharing_options` is more of a json field. When it doesn't exist, it would always grab the default value. Maybe I am wrong about that.